### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by samba2**
two areas of changes:
- removed unnecessary exception handling in CryptoInitializationVector.getBytes() by using StandardCharsets.UTF_8
- applied several cleanups based on Intellij code inspection (boy scout)

### CODE REVIEW

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java [hunk 1]
```diff
@@ -57,8 +55,7 @@
 		private JavaType contextualType;
 
 		@Override
-		public CryptoContainer<?> deserialize(JsonParser jp, DeserializationContext ctxt)
-				throws IOException, JsonProcessingException {
+		public CryptoContainer<?> deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
```
Comment (uweschaefer) [inline cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:58]: looks like we're using different formatting standards here (line length)

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java [hunk 2]
```diff
@@ -129,25 +125,25 @@ protected CryptoContainer(@NonNull Class<T> class1, @NonNull CryptoAlgorithm alg
 	}
 
 	@Getter
-	private Class<?> type;
+	private final Class<?> type;
```
Comment (uweschaefer) [inline cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:128]: not sure if this is semantically correct on a serialized/deserialized object

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java (hunk 1 above)
Comment (samba2) [reply cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:58]: hm, I was  only using spotless to auto-format on commit. I think the new version is just shorter so it removed the extra line.

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoInitializationVector.java [hunk 3]
```diff
@@ -27,18 +27,13 @@
...
+			sb.append(initVector);
 
-			byte[] bytes = sb.toString().getBytes("UTF-8");
-			// take the first 16 bytes
-			return Arrays.copyOf(bytes, 16);
-		} catch (UnsupportedEncodingException e) {
-			// must not happen
-			throw new IllegalStateException("UTF-8 not a valid charset!?");
-		}
+		byte[] bytes = sb.toString().getBytes(StandardCharsets.UTF_8);
```
Comment (uweschaefer) [inline cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoInitializationVector.java:35]: good one, i wasn't aware of that, but the exception bothered me...

File: cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/CryptoAlgorithmTest.java [hunk 4]
```diff
@@ -7,10 +7,8 @@
 public class CryptoAlgorithmTest {
 
 	@Test
-	public void testOf() throws Exception {
-		assertThrows(NullPointerException.class, () -> {
-			CryptoAlgorithm.of(null);
-		});
+	public void testOf()  {
+		assertThrows(NullPointerException.class, () -> CryptoAlgorithm.of(null));
```
Comment (uweschaefer) [inline cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/CryptoAlgorithmTest.java:11]: different settings here.

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java (hunk 2 above)
Comment (samba2) [reply cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:128]: if in doubt I'd change it back then
```suggestion
	private Class<?> type;
```

File: cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/metrics/CryptoMetricsBaseTest.java [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 public class CryptoMetricsBaseTest {
 
-	CryptoMetrics.Base uut = new CryptoMetrics.Base() {
+	final CryptoMetrics.Base uut = new CryptoMetrics.Base() {
```
Comment (uweschaefer) [inline cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/metrics/CryptoMetricsBaseTest.java:9]: suggestion: can we please agree to not do that on the test scope automatically?
i personally like the test scope using as little words as possible in order to make it easy to read.

Comment (samba2) [reply cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/metrics/CryptoMetricsBaseTest.java:9]: I see. need to tell this IDEA then. but I get your point. let it be this then:
```suggestion
	CryptoMetrics.Base uut = new CryptoMetrics.Base() {
```

File: cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/CryptoAlgorithmTest.java (hunk 4 above)
Comment (samba2) [reply cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/CryptoAlgorithmTest.java:11]: hm, not sure what to do about this

**Comment by uweschaefer** [comment]
created a followup ticket #134

File: cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/metrics/CryptoMetricsBaseTest.java (hunk 5 above)
Comment (uweschaefer) [reply cryptoshred-core/src/test/java/eu/prismacapacity/cryptoshred/core/metrics/CryptoMetricsBaseTest.java:9]: you can scope inspections in idea, which is actually a great feature that eclipse is lacking

File: cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java (hunk 1 above)
Comment (uweschaefer) [reply cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:58]: there are no spotless rules in this project, so... hence #134

Comment (samba2) [reply cryptoshred-core/src/main/java/eu/prismacapacity/cryptoshred/core/CryptoContainer.java:58]: Is this better for you ? :
```suggestion
		public CryptoContainer<?> deserialize(JsonParser jp, DeserializationContext ctxt)
		                throws IOException {
```

**Comment by uweschaefer** [comment]
@samba2 finally, i understood, why there is no build for this branch :older_man: 

i just invited you to the team, please feel free to open new branches within the target repos.

**Pull request merged**